src/DssSpellCollateralOnboarding.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// Copyright (C) 2021 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+import "dss-exec-lib/DssExecLib.sol";
+
+contract DssSpellCollateralOnboardingAction {
```

Comment:
We should consider moving the `DssSpellCollateralOnboarding.sol` into a `template` folder, following a similar pattern as was used initially for the [collateral onboarding](https://collateral.makerdao.com/smart-contracts-domain/untitled#initialize-spell-action) spell crafting (before adopting dss-exec-lib), adding detailed docs about the process in the readme, removing related comments in the spell code and consider adding a flag to the archive script to include the file only when there is a collateral onboarding.
cc: @brianmcmichael

Reply:
We should also consider adding a template for RWAs as well.

Reply:
Happy to discuss this, but this feels like it could be its own PR after this one especially since we would probably want a DssSpell template in addition to the Collateral and RWAs.  Agree on the archiving, but I think I would just don't archive the template directory

Reply:
I agree on having a separate PR for RWAs and DssSpell templates eventually.
Regarding the current changes I would suggest to:
- move `DssSpellCollateralOnboarding.sol` (to be copied in `src` when used and archived as well) into a `template` folder, to locate in the repo root folder.
- cleanup the comments in `DssSpell.sol` relative to collateral onboarding, to avoid having extra comments in spell code.
- document the collateral onboarding steps in the readme.
- consider either adding a flag in the archive script or comment that line and add a note in the readme, to avoid copying in the archive an empty file when there is not an onboarding.

Ref. https://github.com/makerdao/spells-kovan/blob/master/template/SpellAction.sol

Reply:
We could move the collateral onboarding changes to a separate PR (along with RWAs and DssSpell templates)  and focus on spell clean for the current one.

Reply:
#212

Reply:
handled in #212